File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func dfs(_ numbers: [Int], _ target: Int, _ index: Int, _ currentSum: Int) -> Int {
    if index == numbers.count {
        return currentSum == target ? 1 : 0
    }    
    var add = dfs(numbers, target, index + 1, currentSum + numbers[index])
    var sub = dfs (numbers, target, index + 1, currentSum - numbers[index])    
    return add + sub
    
}

func solution(_ numbers:[Int], _ target:Int) -> Int {    
    let answer:Int = dfs(numbers, target, 0 , 0)
    return answer    
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    
    function DFS(level, sum) {        
        if(level === numbers.length) {
            if (sum === target) answer++            
        } else {
            DFS(level+1, sum + numbers[level])
            DFS(level+1, sum - numbers[level])
        }       
    }
    DFS(0,0)
    return answer;
}